core/src/main/java/hudson/model/Queue.java

```diff
@@ -1443,9 +1447,11 @@ public void maintain() {
                 }
                 // pending -> buildable
                 for (BuildableItem p: lostPendings) {
-                    LOGGER.log(Level.FINE,
+                    if (LOGGER.isLoggable(Level.FINE)) {
```

Comment:
Note that there are more cases of this optimization that could be done; I just picked some sample cases.